lib/faraday/middleware.rb

```diff
@@ -9,5 +9,9 @@ class Middleware
     def initialize(app = nil)
       @app = app
     end
+
+    def close
+      @app&.close
```

Comment:
```suggestion
      @app.close if @app.respond_to?(:close)
```

This will prevent it from blowing up when it gets to the Adapter. However, it will still call `Adapter#close` if it's implemented in a specific Faraday adapter.